gamification-evm-services/src/main/java/io/meeds/evm/gamification/service/EvmTriggerService.java

```diff
@@ -87,7 +87,8 @@ private void processEvent(EvmTrigger evmTrigger) {
                               + Utils.MIN_AMOUNT + ": " + evmTrigger.getAmount() + ", "
                               + Utils.SENT_DATE + ": " + evmTrigger.getSentDate() + ", "
                               + Utils.TOKEN_BALANCE + ": " + evmTrigger.getTokenBalance() + ", "
-                              + Utils.DURATION + ": " + evmTrigger.getDuration() + "}";
+                              + Utils.DURATION + ": " + evmTrigger.getDuration() + ", "
+                              + Utils.RULE_ID + ": " + evmTrigger.getRuleId() + "}";
```

Comment:
Could you clarify why we need the rule id for the verification please.

Reply:
we added the rule id to obtain a realization per action (filtered in the validation event in the realization service)
Example: if we have 3 hold event actions using the same token address, but the minimum amount is different, here in the realization service, it will create 3 realizations per action, which is not the case; we must obtain one realization per action.

Reply:
You can verify the network ID and contract address being listened to, and then determine the list of events that should be triggered along with the necessary trigger details here [WalletCheckForHoldEventListener.java](https://github.com/Meeds-io/gamification-evm/blob/feature/meedsv2/gamification-evm-services/src/main/java/io/meeds/evm/gamification/listener/WalletCheckForHoldEventListener.java#L75) instead of sending event for ech rule